pyproject.toml

```diff
@@ -313,3 +314,32 @@ ignore_directives = [
     "tab-set",
 ]
 ignore_messages = "^.*(Unknown target name|Undefined substitution referenced)[^`]*$"
+
+[tool.mypy]
+ignore_errors = true
+scripts_are_modules = true
+
+[[tool.mypy.overrides]]
+module = ['mne.evoked', 'mne.io']
+ignore_errors = false
+# Ignore "attr-defined" until we fix stuff like:
+# - BunchConstNamed: '"BunchConstNamed" has no attribute "FIFFB_EVOKED"'
+# - Missing __all__: 'Module "mne.io.snirf" does not explicitly export attribute "read_raw_snirf"'
+# Ignore "no-untyped-call" until we fix stuff like:
+# - 'Call to untyped function "end_block" in typed context'
+# Ignore "no-untyped-def" until we fix stuff like:
+# - 'Function is missing a type annotation'
+# Ignore "misc" until we fix stuff like:
+# - 'Cannot determine type of "_projector" in base class "ProjMixin"'
+# Ignore "assignment" until we fix stuff like:
+# - 'Incompatible types in assignment (expression has type "tuple[str, ...]", variable has type "str")'
+# Ignore "operator" until we fix stuff like:
+# - Unsupported operand types for - ("None" and "int")
+disable_error_code = [
+    'attr-defined',
+    'no-untyped-call',
+    'no-untyped-def',
+    'misc',
+    'assignment',
+    'operator',
```

Comment:
@larsoner I don't need to add a single one of these to successfully run `mypy mne/` without errors. Did you only add those in anticipation of future errors? I'd rather remove or comment out anything we don't need right now.

Reply:
Ah, you were running it with `--strict`. Why?

I don't think we should run with `--strict` for now.

Reply:
To me it's better to know and be explicit here but we can remove it if you want

Reply:
(I'd like to keep the comments though because they do indicate things we can work on in the future)